plaidml/bridge/openvino/ops/prior_box.cpp

```diff
@@ -0,0 +1,412 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "plaidml_ops.hpp"
+#include "plaidml_util.hpp"
+
+#include "ngraph/opsets/opset.hpp"
+#include "ngraph/opsets/opset4.hpp"
+
+#include "plaidml/op/op.h"
+
+using namespace plaidml;  // NOLINT[build/namespaces]
+using ngraph::opset4::PriorBox;
+
+namespace PlaidMLPlugin {
+
+class PriorBoxImpl {
+ public:
+  PriorBoxImpl(const Context& context, const ngraph::op::PriorBoxAttrs& attributes);
+
+  // Fixed size path uses fixed_size, density, fixed_ratio | aspect_ratio
+  void processFixedSizePath();
+
+  // Min size path uses min_size, max_size, aspect_ratio, scale_all_sizes
+  void processMinSizePath();
+
+  // Append variance after former result
+  void processVariance();
+
+  edsl::Tensor& getOutput() { return C_out; }
+
+ private:
+  // Preprocess data with some attributes
+  void prepareConfig();
+
+  // Produce a tensor which data are interlaced with W and H
+  // Use this to produce box center for different path
+  void produceCenterBase();
+
+  const Context& ctx;
+  const ngraph::op::PriorBoxAttrs& attrs;
+  DType precision;
+  int H;
+  int W;
+  edsl::Tensor IH;
+  edsl::Tensor IW;
+  int num_of_prior;
+  int min_element_size;
+  edsl::Tensor Step;
+  edsl::Tensor Step_x;
+  edsl::Tensor Step_y;
+  float offset;
+  std::vector<float> aspect_ratios;
+  std::vector<float> aspect_ratios_scale_size;
+  std::vector<float> variance;
+  edsl::Tensor Min_size;
+  edsl::Tensor C_mix;
+  edsl::Tensor C_out;
+};
+
+PriorBoxImpl::PriorBoxImpl(const Context& context, const ngraph::op::PriorBoxAttrs& attributes)
+    : ctx(context), attrs(attributes) {
+  prepareConfig();
+  produceCenterBase();
+}
+
+// Preprocess data with some attributes
+void PriorBoxImpl::prepareConfig() {
+  precision = to_plaidml(ctx.layer->get_output_element_type(0));
+  auto* input_shape_ngraph_op = ngraph::as_type<ngraph::op::Constant>(ctx.layer->get_input_node_ptr(0));
```

Comment:
Please add a check for whether this is a null pointer, and if so, throw an exception indicating that this input must be a constant (similar to other ops that use `as_type<ngraph::op::Constant>`).

Reply:
Done

Reply:
DONE